src/teleop_twist_joy.cpp

```diff
@@ -68,63 +68,77 @@ struct TeleopTwistJoy::Impl
+  double_parameters.push_back({"scale_linear_turbo_x", 1.0, pimpl_->scale_linear_turbo_map["x"]});
+  double_parameters.push_back({"scale_angular_yaw", 0.5, pimpl_->scale_angular_map["yaw"]});
+  double_parameters.push_back({"scale_angular_turbo_yaw", 0.5, pimpl_->scale_angular_turbo_map["yaw"]});
+
+  for(const auto& p : int_parameters) {
+    this->set_parameter_if_not_set(p.name, p.default_value);
+    this->get_parameter(p.name, p.variable);
   }
 
-  // TODO(clalancette): node->get_parameter(s) doesn't seem to
-  // support getting a map of values yet.  Revisit this once it does.
-  // if (nh_param->getParam("axis_angular", pimpl_->axis_angular_map))
-  // {
-  //   nh_param->getParam("axis_angular", pimpl_->axis_angular_map);
-  //   nh_param->getParam("scale_angular", pimpl_->scale_angular_map);
-  //   nh_param->getParam("scale_angular_turbo", pimpl_->scale_angular_turbo_map);
-  // }
-  // else
-  {
-    pimpl_->axis_angular_map["yaw"] = 2;
-    node->get_parameter("axis_angular", pimpl_->axis_angular_map["yaw"]);
-    pimpl_->scale_angular_map["yaw"] = 0.5;
-    node->get_parameter("scale_angular", pimpl_->scale_angular_map["yaw"]);
-    pimpl_->scale_angular_turbo_map["yaw"] = pimpl_->scale_angular_map["yaw"];
-    node->get_parameter("scale_angular_turbo", pimpl_->scale_angular_turbo_map["yaw"]);
+  for(const auto& p : double_parameters) {
+    this->set_parameter_if_not_set(p.name, p.default_value);
+    this->get_parameter(p.name, p.variable);
   }
 
+  auto param_change_callback =
+    [this](std::vector<rclcpp::Parameter> parameters) -> rcl_interfaces::msg::SetParametersResult
+    {
+      rcl_interfaces::msg::SetParametersResult result;
+      result.successful = true;
+      result.reason = "Success";
+      for (const auto& param : parameters) {
+        try {
+          std::string name = param.get_name();
+          auto int_it = std::find_if(int_parameters.begin(), int_parameters.end(), [&name](const JoyParameter<int>& j) -> bool { return name == j.name; });
+          if (int_it != int_parameters.end()) {
+            // This was one of the required parameters, so ensure it isn't
+            // being deleted
+            if (param.get_type() == rclcpp::ParameterType::PARAMETER_NOT_SET) {
+              throw std::runtime_error("Cannot delete required parameter");
+            }
+            int_it->variable = param.as_int();
+          } else {
+            auto double_it = std::find_if(double_parameters.begin(), double_parameters.end(), [&name](const JoyParameter<double>& j) -> bool { return name == j.name; });
+            if (double_it != double_parameters.end()) {
+              // This was one of the required parameters, so ensure it isn't
+              // being deleted
+              if (param.get_type() == rclcpp::ParameterType::PARAMETER_NOT_SET) {
+                throw std::runtime_error("Cannot delete required parameter");
+              }
+            }
+            double_it->variable = param.as_double();
```

Comment:
This means that any non double or known int parameter will be rejected. Is it intended?
e.g. this means that setting use_sim_time will now be impossible

Reply:
No, that was just a bug, should be fixed now.